docs/tasks/stateful-set/upgrade-from-petsets-to-stateful-sets.md

```diff
@@ -0,0 +1,92 @@
+
+{% endcapture %}
+
+{% capture steps %}
+
+### Differences between alpha Pet Sets and beta Stateful Sets
+
+Pet Set was introduced as an alpha resource in Kubernetes release 1.3, and was renamed to Stateful Set as a beta resource in 1.5. 
+Here are some notable changes:
+
+* **Stateful Set is the new Pet Set**: Pet Set is no longer available in any Kubernetes release >= 1.5. It becomes beta Stateful Set.
+* **Stateful Set fencing**: Fencing guards is implemented in >= 1.5 to prevent split brain scenarios for Stateful Sets.
+* **Flipped debug hook behavior**: The behavior of debug hook for Pet Set, i.e. annotation `pod.alpha.kubernetes.io/initialized`, is now flipped for Stateful Sets. The absence of this annotation will pause the Pet Sets, but will NOT pause the Stateful Sets. 
+
+
+### Upgrading from Pet Sets to Stateful Sets
+
+#### Find all Pet Sets and their cofig files 
+
+First, find all existing Pet Sets in your cluster:
+
+```shell
+$ kubectl get petsets --all-namespaces
+```
+
+If you don't find any existing Pet Sets, you can safely upgrade your cluster to >= 1.5.
+
+If you find existing Pet Sets and you have all their config files at hand, you can continue to the next step to delete those Pet Sets. 
+Otherwise, you need to save their config so that you can recreate them as Stateful Sets later. 
+
+Here's an example command for you to save all existing Pet Sets as one config file. 
+
+```shell
+# Save all existing Pet Sets in all namespaces in a single file. Only needed when you don't have their config files at hand. 
+$ kubectl get petsets --all-namespaces -o yaml > all-petsets.yaml
+```
+
+#### Delete all Pet Sets, non-cascadingly 
+
+If you find existing Pet Sets instead, you need to delete all Pet Sets *non-cascadingly*. You can do this from `kubectl` with `--cascade=false`. 
+Note that if the flag wasn't set, cascading deletion will be performed by default. 
+
+```shell
+# Delete all existing Pet Sets 
+$ kubectl delete petsets -f <pet-set-file> --cascade=false
+
+# Alternatively, delete them by name and namespace
+$ kubectl delete petsets <pet-set-name> -n=<pet-set-namespace> --cascade=false
+
+# Get all Pet Sets again to make sure you deleted them all 
+$ kubectl get petsets --all-namespaces
+```
+
+At this moment, you've deleted all Pet Sets in your cluster, but not their pods, persistent volumes, or persistent volume claims. 
+
+#### Upgrade your Kubernetes cluster to >= 1.5
+
+Now, you may [upgrade your Kubernetes cluster](/docs/admin/cluster-management/#upgrading-a-cluster), or just your master, to >= 1.5.0.
+
+#### Create Stateful Sets
+
+Now, you want to recreate all Pet Sets, but turning them into Stateful Sets. To do this, for every Pet Set config file you have, 
+change `apiVersion` from `apps/v1alpha1` to `apps/v1beta1` and change `kind` from `PetSet` to `StatefulSet`. 
```

Comment:
On a practical level, it's probably best to prepare these updated files before actually deleting the petsets, to minimize downtime of the controller. I'm not sure what's the best place to mention that. Would it make sense to describe the file modifications as the first step up above, when you say to either gather or download the YAML?